Support module-path

I don't like the relative behavior of the require of node, hence, I'm using
```js
// Set the /src folder as root
const path = require('path');
require('app-module-path').addPath(path.join(__dirname, ''));
```
to change the root of my import to the `/src` folder of my project.

In order to make this plugin works, I had to change this [line](https://github.com/istarkov/babel-plugin-webpack-loaders/blob/master/src/plugin.js#L100) with:
```js
filePath = '/Users/oliviertassinari/MyProject/src/' + filePath;
```
Any idea of a better solution so that I don't have to do this ugly fix :see_no_evil:?

getting simular issue:

```bash
Error: /Users/ph185068/projects/di/leapfrog/poc/leapfrog-mono-repo/src/App.js: Cannot find module '/Users/ph185068/projects/di/leapfrog/poc/leapfrog-mono-repo/src/__tests__/webpack.config.dev.js'
```

```javascript
{
  "presets": ["react", "es2015"],
  "env": {
    "development": {
      "presets": ["react-hmre"]
    },
    "production": {
      "plugins": [
        [
          "babel-plugin-webpack-loaders",
          {
            "config": "./webpack.config.dev.js",
            "verbose": false,
          }
        ]
      ]
    }
  }
}
```

Could you give a bit more detail on `resolve`

--- a/.babelrc
+++ b/.babelrc
@@ -1,5 +1,5 @@
 {
-  "presets": ["es2015"],
+  "presets": ["es2015", "stage-0"],
   "env": {
     "EXAMPLES_LIB": {
       "plugins": [

--- a/.eslintrc
+++ b/.eslintrc
@@ -1,33 +1,12 @@
 {
-  "extends": "eslint-config-airbnb",
+  "extends": "airbnb",
+  "parser": "babel-eslint",
   "env": {
     "browser": true,
     "mocha": true,
     "node": true
   },
-  "globals": {
-    "__DEV__": false,
-    "__DEV_TOOLS__": false,
-    "ReactElement": false,
-    "ReactComponent": false,
-  },
   "rules": {
-    "max-len": [1, 100, 2],
-    "id-length": 0,
-    "react/no-multi-comp": 0,
-    "react/jsx-uses-react": 2,
-    "react/jsx-uses-vars": 2,
-    "react/react-in-jsx-scope": 2,
-    "babel/generator-star-spacing": 1,
-    "babel/new-cap": 1,
-    "babel/object-shorthand": 1,
-    "babel/arrow-parens": 1,
-    "babel/no-await-in-loop": 1,
-    "block-scoped-var": 0,
-    "padded-blocks": 0
-  },
-  "plugins": [
-    "react",
-    "babel"
-  ]
+    "no-nested-ternary": 0
+  }
 }

--- a/package.json
+++ b/package.json
@@ -28,12 +28,13 @@
   ],
   "dependencies": {
     "babel-preset-es2015": "^6.3.13",
+    "babel-preset-stage-0": "^6.5.0",
     "babel-register": "^6.4.3",
     "babel-traverse": "^6.3.26",
     "babel-types": "^6.3.24",
     "babylon": "^6.3.26",
     "colors": "^1.1.2",
-    "resolve": "^1.1.6",
+    "enhanced-resolve": "^2.2.2",
     "rimraf": "^2.5.0",
     "shell-quote": "^1.4.3",
     "webpack": "^1.12.9"

--- a/src/plugin.js
+++ b/src/plugin.js
@@ -1,5 +1,5 @@
-import { resolve as fsResolve, dirname } from 'path';
-import resolve from 'resolve';
+import { resolve, dirname } from 'path';
+import { ResolverFactory, SyncNodeJsInputFileSystem } from 'enhanced-resolve';
 import { parse } from 'babylon';
 import traverse from 'babel-traverse';
 import runWebPackSync from './runWebPackSync';
@@ -60,7 +60,7 @@ export default function ({ types: t }) {
   // some strange error occurs if I move this fn outside context
   // and then run babel-node with this plugin
   const localInteropRequire = (path) => {
-    const res = require(fsResolve(process.cwd(), path));
+    const res = require(resolve(process.cwd(), path));
     if ('default' in res) {
       return res.default;
     }
@@ -82,7 +82,7 @@ export default function ({ types: t }) {
           return;
         }
 
-        const config = localInteropRequire(fsResolve(process.cwd(), configPath));
+        const config = localInteropRequire(resolve(process.cwd(), configPath));
         if (Object.keys(config).length === 0) {
           // it's possible require calls inside webpack config or bad config
           return;
@@ -92,12 +92,12 @@ export default function ({ types: t }) {
         if (config.module.loaders.some((l) => l.test.test(args[0].value))) {
           const [{ value: filePath }] = args;
 
-          const fileAbsPath = resolve.sync(
-            filePath,
-            {
-              basedir: dirname(filenameRelative),
-            }
-          );
+          const resolver = ResolverFactory.createResolver({
+            fileSystem: new SyncNodeJsInputFileSystem(),
+            ...config.resolve,
+          });
+
+          const fileAbsPath = resolver.resolveSync({}, dirname(filenameRelative), filePath);
 
           const webPackResult = runWebPackSync({ path: fileAbsPath, configPath, config, verbose });
 

--- a/test/runtime.spec.js
+++ b/test/runtime.spec.js
@@ -43,4 +43,9 @@ describe('runtime test', () => {
     const text = require('normalize.css/normalize.css');
     expect(text).toEqual(undefined);
   });
+
+  it('resolve.modules from webpack config should work', () => {
+    const text = require('assets/file.txt');
+    expect(text).toEqual('file.txt');
+  });
 });

--- a/test/runtime.webpack.config.js
+++ b/test/runtime.webpack.config.js
@@ -13,6 +13,12 @@ module.exports = {
   postcss: [
     autoprefixer({ browsers: ['last 2 versions'] }),
   ],
+  resolve: {
+    modules: [
+      __dirname,
+      'node_modules',
+    ],
+  },
   module: {
     loaders: [
       {

